File: lib/models/time_slot.dart
class TimeSlot {
  final String id;
  final String restaurantId;
  final DateTime date;
  final String startTime;
  final String endTime;
  final int capacity;
  final int currentOrders;
  final bool isAvailable;

  TimeSlot({
    required this.id,
    required this.restaurantId,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.capacity,
    this.currentOrders = 0,
    this.isAvailable = true,
  });

  factory TimeSlot.fromJson(Map<String, dynamic> json) {
    return TimeSlot(
      id: json['_id'] ?? '',
      restaurantId: json['restaurantId'] ?? '',
      date: DateTime.parse(json['date'] ?? DateTime.now().toString()),
      startTime: json['startTime'] ?? '',
      endTime: json['endTime'] ?? '',
      capacity: json['capacity'] ?? 20,
      currentOrders: json['currentOrders'] ?? 0,
      isAvailable: json['isAvailable'] ?? true,
    );
  }

  bool get isFull => currentOrders >= capacity;

  String get displayTime => '$startTime - $endTime';

  Map<String, dynamic> toJson() => {
        '_id': id,
        'restaurantId': restaurantId,
        'date': date.toIso8601String(),
        'startTime': startTime,
        'endTime': endTime,
        'capacity': capacity,
        'currentOrders': currentOrders,
        'isAvailable': isAvailable,
      };
}

File: backend/models/TimeSlot.js
const mongoose = require('mongoose');

const timeSlotSchema = new mongoose.Schema(
  {
    restaurantId: { type: String, required: true }, // Foreign key to Restaurant
    date: { type: Date, required: true }, // Date of the time slot
    startTime: { type: String, required: true }, // e.g., "10:00"
    endTime: { type: String, required: true }, // e.g., "10:15"
    capacity: { type: Number, required: true }, // Max orders allowed in this slot
    currentOrders: { type: Number, default: 0 }, // Current orders in this slot
    isAvailable: { type: Boolean, default: true }, // Whether slot is open for orders
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
  },
  { timestamps: true }
);

// Compound index to ensure unique slot per restaurant per date/time
timeSlotSchema.index({ restaurantId: 1, date: 1, startTime: 1 }, { unique: true });

module.exports = mongoose.model('TimeSlot', timeSlotSchema);
